octopoes/octopoes/core/service.py

```diff
@@ -178,6 +178,15 @@ def save_origin(
                 logger.debug("Affirmation source %s already deleted", origin.source)
                 return
 
+        if (
+            origin.origin_type == OriginType.AFFIRMATION
+            and [origin.source] == origin.result
+            and not self.origin_repository.list_origins(valid_time=valid_time, result=origin.source)
```

Comment:
> This will also return affirmation origins, so if there already is an affirmation we won't delete anything. It seems to me that we need to exclude affirmations here.

I do not understand. You mean in the case of a double affirmation -- shouldn't the first one be enough?

> Looking at the code, _delete_ooi already implements the logic we need. The method could have a better name because it doesn't just delete the OOI, but checks if the OOI should be deleted based on the origins and deletes it when there are no origins found that proofs the OOI.

Interesting, why do we end up these dangling creatures?
![image](https://github.com/user-attachments/assets/b1cdd33d-e7ed-4097-a582-75e580f8c1df)